File: arr.c



#include <stdio.h>

int checkSumPair(int arr[], int n, int x) {
    for (int i = 0; i < n - 1; i++) {
        for (int j = i + 1; j < n; j++) {
            if (arr[i] + arr[j] == x) {
                return 1;
            }
        }
    }
    return 0;
}

int main() {
    int arr[] = {0, -1, 42, -3, 1};
    int x = -2;
    int n = sizeof(arr) / sizeof(arr[0]);
    
    if (checkSumPair(arr, n, x)) {
        printf("Yes, there exist two elements in the array that sum up to %d\n", x);
    } else {
        printf("No, there are no two elements in the array that sum up to %d\n", x);
    }
    
    return 0;
}

File: arr.py
"""So Guys welcome you there"""
"""Give an array A[] of n numbers and another number x, the task is to check whether or not there exist two elements in A[] whose sum is exactly x. 

"""

"""Code Starts From Here"""
def check_sum_pair(arr, x):
    seen = set()
    for num in arr:
        if x - num in seen:
            return True
        seen.add(num)
    return False

arr = [0, -1, 42, -3, 1]
x = -2
print(check_sum_pair(arr, x))

"""Code Ends"""
